CMakeLists.txt

```diff
@@ -800,6 +800,21 @@ endif()
 # Should we cross-compile the standard library for Android?
 is_sdk_requested(ANDROID swift_build_android)
 if(swift_build_android AND NOT "${SWIFT_ANDROID_NDK_PATH}" STREQUAL "")
+  # Get the prebuilt suffix to create the correct toolchain path when using the NDK
+  if("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
+    set(_swift_android_prebuilt_suffix "darwin-x86_64")
+  elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
+    set(_swift_android_prebuilt_suffix "linux-x86_64")
+  endif()
+
+  set(SWIFT_ANDROID_PREBUILT_PATH
+      "${SWIFT_ANDROID_NDK_PATH}/toolchains/arm-linux-androideabi-${SWIFT_ANDROID_NDK_GCC_VERSION}/prebuilt/${_swift_android_prebuilt_suffix}")
+  
+  # Resolve the correct linker based on the file name of CMAKE_LINKER (being 'ld' or 'ld.gold' the options)
+  get_filename_component(CMAKE_LINKER_NAME "${CMAKE_LINKER}" NAME)
```

Comment:
Correct, it does. The result is being saved in `CMAKE_LINKER_NAME` (this is not a CMAKE var, so I can change it to `SWIFT_ANDROID_LINKER_NAME`). The params of [`get_filename_component`](https://cmake.org/cmake/help/v3.0/command/get_filename_component.html) are `<DEST_VAR> <FILE_NAME> <COMPONENT>`.